contrib/macdeploy/README.md

```diff
@@ -14,55 +14,44 @@ When complete, it will have produced `Bitcoin-Qt.dmg`.
 
 ## SDK Extraction
 
-Our current macOS SDK (`macOSX10.14.sdk`) can be extracted from
-[Xcode_10.2.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_10.2.1/Xcode_10.2.1.xip).
+### Step 1: Obtaining `Xcode.app`
+
+Our current macOS SDK
+(`Xcode-11.3.1-11C505-extracted-SDK-with-libcxx-headers.tar.gz`) can be
+extracted from
+[Xcode_11.3.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_11.3.1/Xcode_11.3.1.xip).
```

Comment:
>  Also, maybe use 11.5?

NACK. 11.3.1 was picked specifically. It's what is referenced throughout the docs, build  system and  in the tarball name. There's also no guarantee that you'll generate the same tarball between various versions of Xcode.

Reply:
What do you mean by "picked specifically"? It's introduced in this PR. The only other place I see it is on the Travis macOS native machine, but the choice there doesn't matter.

Reply:
Sorry for not explaining in more detail in the commit message.

11.3.1 was chosen as the last Xcode version which uses (Apple's version of) clang 8.x.x, for which we have a matching cctools and ld64. Starting with 11.4, they bumped to clang 9.x.x. Once cctools-port has a working matching version for clang 9, we can bump to a newer Xcode version.

Reply:
A thought: to prevent future confusion, we should follow up this PR with documentation on the reasoning behind how our macOS toolchain is constructed.

Reply:
More info would be a good followup. I'm surprised Apple would bump clang 8 to 9 in a minor Xcode version update. But actually that could make sense, because they want to give developers access to SDK's for their next generation(s) OS, without forcing devs to upgrade their production machine to a beta/recent macOS. According to [wikipedia](https://en.wikipedia.org/wiki/Xcode#11.x_series) Xcode 11.3.1 requires macOs 10.14.4, whereas Xcode 11.4 bumps that requirement to macOS 10.15.2. The specific choice of clang 8.x.x makes sense.